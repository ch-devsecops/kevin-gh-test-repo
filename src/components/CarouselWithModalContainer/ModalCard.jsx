import React from 'react';
import PropTypes from 'prop-types';
import { Wrapper, Box, Markdown, MarkdownHeading, Media, Image } from '@honda-canada/design-system-react';
import themeStyles from './ModalCard.styles';

const Container = themeStyles.apply(Wrapper, 'Container');
const Content = themeStyles.apply(Box, 'Content');
const Title = themeStyles.apply(MarkdownHeading, 'Title');
const BodyText = themeStyles.apply(Markdown, 'BodyText');
const StyledMedia = themeStyles.apply(Media, 'Media');
const MediaContainer = themeStyles.apply(Box, 'MediaContainer');

// TODO: Evaluate moving this component to DSR
const OptionalImage = ({ image, ...rest }) => (
  <StyledMedia {...rest}>
    <MediaContainer>
      <Image {...image} />
    </MediaContainer>
  </StyledMedia>
);

OptionalImage.propTypes = {
  image: PropTypes.shape({
    src: PropTypes.string,
    alt: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
  }),
};

const ModalCard = ({ title, bodyText, image }) => (
  <Container>
    <OptionalImage greaterThanOrEqual="desktop" image={image} />
    <Content>
      <Title headingOverride="h3">{title}</Title>
      <OptionalImage lessThan="desktop" image={image} />
      <BodyText>{bodyText}</BodyText>
    </Content>
  </Container>
);

ModalCard.propTypes = {
  title: PropTypes.string,
  bodyText: PropTypes.string,
  image: PropTypes.shape({
    src: PropTypes.string,
    alt: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
  }),
};

export default ModalCard;
