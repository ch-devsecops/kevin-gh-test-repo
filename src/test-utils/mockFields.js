/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
