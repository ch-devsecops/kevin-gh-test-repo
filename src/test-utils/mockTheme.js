/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
