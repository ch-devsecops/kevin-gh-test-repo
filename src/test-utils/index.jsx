import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
