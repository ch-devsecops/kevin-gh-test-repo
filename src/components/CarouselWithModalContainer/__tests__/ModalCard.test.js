import React from 'react';
import { jssRender, cleanup, axe } from 'test-utils';
import ModalCard from '../ModalCard';

afterEach(cleanup);

const mockData = {
  title: 'Modal Title',
  bodyText: 'Modal body text',
  image: {
    src: './sample-mock-data/media/img/placeholders/placeholder-landscape.png',
    alt: 'Unsplash image',
  },
};

describe('<ModalCard>', () => {
  it('should render correctly', () => {
    const { container } = jssRender(<ModalCard {...mockData} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with empty props', () => {
    const { container } = jssRender(<ModalCard />);

    expect(container).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const { container } = jssRender(<ModalCard {...mockData} />);

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
