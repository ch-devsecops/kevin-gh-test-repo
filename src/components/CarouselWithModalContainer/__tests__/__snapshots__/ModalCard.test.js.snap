// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalCard> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sc-1k44dfv-0 fOlbHO"
  >
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-desktop sc-1k44dfv-0 bCQNSY"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
      >
        <img
          alt="Unsplash image"
          class="sc-1k44dfv-0 jLTFZk"
          draggable="true"
          loading="lazy"
          src="./sample-mock-data/media/img/placeholders/placeholder-landscape.png"
        />
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 tDSQa"
    >
      <div
        class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 ApqDi"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Modal Title
        </h3>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-desktop sc-1k44dfv-0 bCQNSY"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
        >
          <img
            alt="Unsplash image"
            class="sc-1k44dfv-0 jLTFZk"
            draggable="true"
            loading="lazy"
            src="./sample-mock-data/media/img/placeholders/placeholder-landscape.png"
          />
        </div>
      </div>
      <div
        class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 csdcDA"
      >
        

        <p>
          Modal body text
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ModalCard> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sc-1k44dfv-0 fOlbHO"
  >
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-desktop sc-1k44dfv-0 bCQNSY"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
      >
        <img
          alt="Unsplash image"
          class="sc-1k44dfv-0 jLTFZk"
          draggable="true"
          loading="lazy"
          src="./sample-mock-data/media/img/placeholders/placeholder-landscape.png"
        />
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 tDSQa"
    >
      <div
        class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 ApqDi"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Modal Title
        </h3>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-desktop sc-1k44dfv-0 bCQNSY"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
        >
          <img
            alt="Unsplash image"
            class="sc-1k44dfv-0 jLTFZk"
            draggable="true"
            loading="lazy"
            src="./sample-mock-data/media/img/placeholders/placeholder-landscape.png"
          />
        </div>
      </div>
      <div
        class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 csdcDA"
      >
        

        <p>
          Modal body text
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ModalCard> should render correctly with empty props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sc-1k44dfv-0 fOlbHO"
  >
    <div
      class="fresnel-container fresnel-greaterThanOrEqual-desktop sc-1k44dfv-0 bCQNSY"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
      >
        <img
          class="sc-1k44dfv-0 jLTFZk"
          draggable="true"
          loading="lazy"
        />
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 tDSQa"
    >
      <div
        class="fresnel-container fresnel-lessThan-desktop sc-1k44dfv-0 bCQNSY"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 iTfAOr"
        >
          <img
            class="sc-1k44dfv-0 jLTFZk"
            draggable="true"
            loading="lazy"
          />
        </div>
      </div>
      <div
        class="sc-1k44dfv-0 klHKZs sc-1k44dfv-0 csdcDA"
      >
        

        <p>
          undefined
        </p>
      </div>
    </div>
  </div>
</div>
`;
